### DISCUSSION TITLE
Add seaweedfs volume to a current seaweedfs docker setup

### DISCUSSION THREAD

**Discussion by jhill2727** [seaweedfs/seaweedfs#3910] [Q&A] [answered]
We currently have a seaweedfs server running master, volume, and filer containers. We would like to spin up another server (or several, eventually using kubernetes to organize) with just a volume container for the master and filer to use. I've tried a couple of docker-compose configurations, and they don't seem to be what it's looking for. Is there a specific page in the wiki for getting this going? I've seen how to configure a single master/filer and multiple volumes as a service, but I can't quite seem to translate that to running them as containers. I've tried a few things in the compose file on the new volume server, and will gladly share them if needed. That makes the post a bit lengthy, though, so is there a documented way of configuring this to work?

**Comment by chrislusf** [comment]
checked this? https://github.com/seaweedfs/seaweedfs/blob/master/docker/compose/local-cluster-compose.yml

**Reply by jhill2727** [reply]
I had not seen that yet, but that seems to spin up multiple volumes on the same server using localhost but pointing to different DCs and racks. I'm looking for a volume on a separate volume server using a master on a separate master server. The closest I've gotten is doing the following command in the compose on the volume server:

'volume -mserver="172.17.2.181:9333" -port=8080 -max=0 -minFreeSpacePercent=1 dataCenter=dc1 -rack=rack1 -dir=/data/seaweedfs/volume publicUrl=172.17.2.142:8080 ip=172.17.2.142

This results in the following logs while starting up (note that container master:9333 is running on 172.17.2.181):

volume_1  | I1026 14:29:44     1 file_util.go:23] Folder /data Permission: -rwxrwxr-x
volume_1  | I1026 14:29:44     1 disk_location.go:209] Store started on dir: /data with 0 volumes max 0
volume_1  | I1026 14:29:44     1 disk_location.go:212] Store started on dir: /data with 0 ec shards
volume_1  | I1026 14:29:44     1 volume_grpc_client_to_master.go:52] Volume server start with seed master nodes: [172.17.2.181:9333]
volume_1  | I1026 14:29:44     1 volume.go:364] Start Seaweed volume server 30GB 3.06 2f846777 at 172.22.0.2:8080
volume_1  | I1026 14:29:44     1 volume_grpc_client_to_master.go:109] Heartbeat to: 172.17.2.181:9333
volume_1  | I1026 14:29:44     1 volume_grpc_client_to_master.go:143] Volume Server found a new master newLeader: master:9333 instead of 172.17.2.181:9333
volume_1  | I1026 14:29:47     1 volume_grpc_client_to_master.go:106] SendHeartbeat to master:9333: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: lookup master on 127.0.0.11:53: server misbehaving"
volume_1  | I1026 14:29:47     1 volume_grpc_client_to_master.go:71] heartbeat error: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: lookup master on 127.0.0.11:53: server misbehaving"

**Reply by chrislusf** [reply]
seems a docker network issue. you would need to replace "master" with the actual host server IP address.

**Reply by jhill2727** [reply]
On which side would I replace? On the volume server side it has the IP of 172.17.2.181. On the master side it can't bind if I change it from ip=master

master_1  | I1031 12:46:13     1 file_util.go:23] Folder /data Permission: -rwxr-xr-x
master_1  | I1031 12:46:13     1 master.go:255] current: 172.17.2.181:9333 peers:
master_1  | I1031 12:46:13     1 master_server.go:131] Volume Size Limit is 1024 MB
master_1  | I1031 12:46:13     1 master.go:147] Start Seaweed Master 30GB 3.06 2f846777 at 172.17.2.181:9333
master_1  | F1031 12:46:13     1 master.go:150] Master startup error: listen tcp 172.17.2.181:9333: bind: cannot assign requested address
volume_1  | I1031 12:46:13     1 volume_grpc_client_to_master.go:43] checkWithMaster 172.17.2.181:9333: get master 172.17.2.181:9333 configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 172.17.2.181:19333: connect: connection refused"

**Comment by jhill2727** [comment, marked as answer]
Got it. I had to enable the master service on the master server to accept external connections on 9333. It now sees a volume child on the volume server. Docker networking....SMH. Thanks!
